operator/attestor/attestor.go

```diff
@@ -200,22 +201,36 @@ func (attestor *Attestor) reconnectClient(rollupId uint32) (eth.Client, error) {
 
 // Spawns routines for new headers that die in one minute
 func (attestor *Attestor) processRollupHeaders(rollupId uint32, headersC chan *ethtypes.Header, subscription ethereum.Subscription, ctx context.Context) {
+	reinitializeTicker := time.NewTicker(REINITIALIZE_DELAY)
+	reinitializeTicker.Stop()
+
+	reinitializeSubscription := func() error {
+		client, err := attestor.reconnectClient(rollupId)
+		if err != nil {
+			attestor.logger.Error("Error while reconnecting client", "rollupId", rollupId, "err", err)
+			return err
+		}
+		attestor.clients[rollupId] = client
```

Comment:
Looking at it now it seems like we could use mutex here. If error happens on our side I guess multiple subscriptions could get it at the same time

Reply:
Yeah, makes sense. Those should be different clients, but it could happen multiple are triggered at once

Reply:
Done in 8d7058c